packages/block-editor/src/store/test/reducer.js

```diff
@@ -1786,196 +1790,150 @@ describe( 'state', () => {
-				type: 'START_MULTI_SELECT',
-			} );
+	describe( 'selectionStart() and selectionEnd()', () => {
+		it( 'should set multi selection', () => {
+			const action = {
+				type: 'MULTI_SELECT',
+				start: 'ribs',
+				end: 'chicken',
+			};
+			const state1 = selectionStart( undefined, action );
+			const state2 = selectionEnd( undefined, action );
 
-			expect( state ).toEqual( {
-				start: { clientId: 'ribs' },
-				end: { clientId: 'ribs' },
-				initialPosition: null,
-				isMultiSelecting: true,
-			} );
+			expect( state1 ).toEqual( { clientId: 'ribs' } );
+			expect( state2 ).toEqual( { clientId: 'chicken' } );
 		} );
 
-		it( 'should return same reference if already multi-selecting', () => {
-			const original = deepFreeze( { start: { clientId: 'ribs' }, end: { clientId: 'ribs' }, isMultiSelecting: true } );
-			const state = blockSelection( original, {
-				type: 'START_MULTI_SELECT',
-			} );
+		it( 'should reset selection', () => {
+			const start = { clientId: 'a' };
+			const end = { clientId: 'b' };
+			const action = {
+				type: 'RESET_SELECTION',
+				selectionStart: start,
+				selectionEnd: end,
+			};
+			const state1 = selectionStart( undefined, action );
+			const state2 = selectionEnd( undefined, action );
 
-			expect( state ).toBe( original );
+			expect( state1 ).toEqual( start );
+			expect( state2 ).toEqual( end );
 		} );
 
-		it( 'should end multi selection with selection', () => {
-			const original = deepFreeze( { start: { clientId: 'ribs' }, end: { clientId: 'chicken' }, isMultiSelecting: true } );
-			const state = blockSelection( original, {
-				type: 'STOP_MULTI_SELECT',
-			} );
+		it( 'should change selection', () => {
+			const action = {
+				type: 'SELECTION_CHANGE',
+				clientId: 'a',
+				attributeKey: 'content',
+				startOffset: 1,
+				endOffset: 2,
+			};
+			const state1 = selectionStart( undefined, action );
+			const state2 = selectionEnd( undefined, action );
 
-			expect( state ).toEqual( {
-				start: { clientId: 'ribs' },
-				end: { clientId: 'chicken' },
-				initialPosition: null,
-				isMultiSelecting: false,
+			expect( state1 ).toEqual( {
+				clientId: 'a',
+				attributeKey: 'content',
+				offset: 1,
 			} );
-		} );
-
-		it( 'should return same reference if already ended multi-selecting', () => {
-			const original = deepFreeze( { start: { clientId: 'ribs' }, end: { clientId: 'chicken' }, isMultiSelecting: false } );
-			const state = blockSelection( original, {
-				type: 'STOP_MULTI_SELECT',
+			expect( state2 ).toEqual( {
+				clientId: 'a',
+				attributeKey: 'content',
+				offset: 2,
 			} );
-
-			expect( state ).toBe( original );
 		} );
+	} );
 
-		it( 'should end multi selection without selection', () => {
-			const original = deepFreeze( { start: { clientId: 'ribs' }, end: { clientId: 'ribs' }, isMultiSelecting: true } );
-			const state = blockSelection( original, {
-				type: 'STOP_MULTI_SELECT',
+	describe( 'selection()', () => {
```

Comment:
Conceptually, should we be testing what is an internal helper for actually `selectionStart` and `selectionEnd`, or should we instead aim those tests at the final reducers? Strictly speaking, by testing `selection` we make no guarantees that `selection{Start,End}` follow that same behaviour.

Reply:
Yes, that's true. We can test `selectionStart` and `selectionEnd` instead...

Reply:
I changed the tests.